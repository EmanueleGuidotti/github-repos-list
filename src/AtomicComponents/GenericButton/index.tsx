import React from "react";
import {Button} from "../../Interfaces/interfaces";

const GenericButton = ({text, onClickCursor}: Button) => {
    return (
        <button
            style={{...componentStyle.button} as React.CSSProperties}
            onClick={() => onClickCursor && onClickCursor()}
        >
            {text}
        </button>
    );
};

const componentStyle = {
    button: {
        padding: "5",
        flex: "1",
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
    },
};

export default GenericButton;
