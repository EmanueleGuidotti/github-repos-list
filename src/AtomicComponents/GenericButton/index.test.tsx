import React from "react";
import renderer from "react-test-renderer";
import GenericButton from "./index";

describe("GenericButtonComponent", () => {
    it("render Generic Button component", () => {
        const component = renderer.create(<GenericButton  text="100" onClickCursor={()=>{}} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
