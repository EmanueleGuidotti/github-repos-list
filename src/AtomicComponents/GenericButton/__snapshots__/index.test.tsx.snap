// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericButtonComponent render Generic Button component 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": "1",
      "justifyContent": "center",
      "padding": "5",
    }
  }
>
  100
</button>
`;
